src/ray/raylet/node_manager.cc

```diff
@@ -697,8 +698,12 @@ void NodeManager::HandleReleaseUnusedBundles(
     auto &worker = worker_it.second;
     const auto &bundle_id = worker->GetBundleId();
     // We need to filter out the workers used by placement group.
-    if (!bundle_id.first.IsNil() && 0 == in_use_bundles.count(bundle_id)) {
-      workers_associated_with_unused_bundles.emplace_back(worker);
+    if (!bundle_id.first.IsNil()) {
+      if (auto iter = in_use_bundles.find(bundle_id.first);
```

Comment:
Just minor, but since it is a bit hard to undertand
```suggestion
    const auto &pg_id = bundle_id.first
    const auto &bundle_index = bundle_id.second
    if (!pg_id.IsNil())
        if (auto iter = in_use_bundles.find(pg_id);
            iter == in_use_bundles.end() || // cannot find the placement group or
            (bundle_index != -1 &is & iter->second.count(bundle_index) == 0) // bundle index is not used (-1 means blah blah)
```